core/src/main/scala/org/apache/spark/deploy/master/SchedulingAlgorithm.scala

```diff
@@ -0,0 +1,171 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.deploy.master
+
+/**
+ * An interface for sort algorithm
+ */
+private[spark] trait SchedulingAlgorithm {
```

Comment:
@CodingCat Thanks for commenting.

In fact, we are going to implement a scheduler for Spark applications in standalone deployment mode. Because of that, we need an extensive interface for implementing new scheduling algorithms. This interface can be static like this PR or dynamic one like I mentioned above that loads scheduling algorithm class in runtime. Both are ok for us. We just want to make sure our implementation can be contributed back later and maintained easily.

This interface in current stage can be not so much comprehensive. It can be revised later. But we want to make sure that there is an extensive interface we can build our scheduling algorithm based on.